ethereumj-core/src/main/java/org/ethereum/net/rlpx/MessageCodesResolver.java

```diff
@@ -34,12 +36,21 @@ public void init(List<Capability> caps) {
         int offset = P2pMessageCodes.USER.asByte() + 1;
 
         for (Capability capability : caps) {
+
             if (capability.getName().equals(Capability.ETH)) {
                 setEthOffset(offset);
                 EthVersion v = fromCode(capability.getVersion());
                 offset += EthMessageCodes.values(v).length;
             }
 
+            if (capability.getName().equals(Capability.PAR)) {
+                // TODO: Why???
+                setParOffset(33);
```

Comment:
@arkpar as you are here, at https://github.com/ethcore/parity/wiki/Warp-Sync it's written that "include snapshot_hash and snapshot_number which signify the block hash and number respectively", but hash is hash of the manifest RLP, and you are validating it by comparison with manifest's RLP hash in Parity's code. Am I correct?